### DISCUSSION TITLE
Persisting many to many relationship object to the database, silent error

### DISCUSSION THREAD

**Discussion by osviel91** [symfony/symfony#46116] [Q&A]
Hi guys i'm having a problem when i'm edditing a entity and save it in the database. 

I have two entities, "area" and "worker", they have a many to many relationships between. When i create a new Worker, i just select the area (or many) and works fine, te area persist in the db. The issue it's when i want to edit the worker the changes in the area don't persist in te db and no error apear.

By the other hand i have another etity who is related with area same way and changes works in the edditing. I have checked the code fully and no differences i can observe.

**Comment by MichaelBrauner** [comment]
Without any code samples it's hard to help you. Can you post your editAction inside your WorkerController?

**Reply by osviel91** [reply]
Ok, if you need anything just ask. Tanks a lot!

**Comment by MichaelBrauner** [comment]
Can you please format it correctly , so that all code is recognized as code? And please make sure you focus on the things related to your topic. (i.e. editing `Worker` - especially changes in the `$area` field are not persisted). Nobody want's to have to dig through a bunch of unrealted code in order to help.

And also make sure you use the same terminology in the code as in your question. Unfortunately, I can't speak Spanish or Portuguese and, to be honest, I don't feel like entering all the terms into a translator to find the lines that are important.

What is missing is your FormType (NOT the whole FormType - just everything related to `Worker` and `Area`) and maybe the specific lines in your template, when you have done something different than `{{ form(form) }}`.

**Comment by osviel91** [comment]
Hi, i'm so sorry i was checking how to format the code for some reason was not working...
My issue es next: When i'm trying to persist a ManyToMany relationship does't work and no error showed up. 
Almos all my code is generated by the sf console. I'll pase here the function to create a new "Worker" entity:

`public function new(Request $request, WorkerRepository $workerRepository, UserPasswordHasherInterface $passwordHasher, SluggerInterface $slugger, ManagerRegistry $doctrine): Response
    {
        
        $worker= new Worker();
        $form = $this->createForm(WorkerType::class, $worker);
        $form->handleRequest($request);

        if ($form->isSubmitted() && $form->isValid()) {

            // hash the password
            $hashedPassword = $passwordHasher->hashPassword(
                $worker,
                $worker->getPassword()
            );

            $worker->setPassword($hashedPassword);

            //-. has the password

            $workerRepository->add($trabajador);

            //Add the return info
            $this->addFlash(
                'notice',
                'Trabajador ' . $worker->getName() . 'Added'
            );
            //-. add the return info

            return $this->redirectToRoute('app_worker_index', [], Response::HTTP_SEE_OTHER);
        }

        return $this->renderForm('worker/new.html.twig', [
            'worker' => $worker,
            'form' => $form,
        ]);
    }`

**Reply by osviel91** [reply]
The relationship it's between Worker and Area. so When i'm creating a new worker and i select the different areas he belong it's not been persisted in the db. But it does work from the Area side.
